sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java

```diff
@@ -73,24 +72,38 @@ public void execute(SensorContext context) {
 
     warnDeprecatedPropertyUsage(config);
 
-    HashSet<InputFile> filesCovered = new HashSet<>();
-    List<File> reports = getCoverageReports(baseDir, config);
-    if (!reports.isEmpty()) {
-      LOG.info("Python test coverage");
-      for (File report : uniqueAbsolutePaths(reports)) {
-        Map<InputFile, NewCoverage> coverageMeasures = parseReport(report, context);
-        saveMeasures(coverageMeasures, filesCovered);
+    try {
+      HashSet<InputFile> filesCovered = new HashSet<>();
+      List<File> reports = getCoverageReports(baseDir, config);
+      if (!reports.isEmpty()) {
+        LOG.info("Python test coverage");
+        for (File report : uniqueAbsolutePaths(reports)) {
+          importReport(context, report, filesCovered);
+        }
       }
+    } catch (Exception e) {
+      LOG.warn("Cannot read coverage report, the following exception occurred: '{}'", e.getMessage());
+      analysisWarnings.addUnique("An error occurred while trying to import coverage report(s)");
+    }
```

Comment:
I guess this part can be reached only if `getCoverageReports` throws and so if `PythonReportSensor.getReports` throws. To test it you could mock a throw from `PythonReportSensor.getReports`, this would prove as well that the sensor will not just crash. This would require a bit of a rework, but this would cover all bases.

Reply:
Oh, I didn't realize that static methods can be mocked